conda_smithy/lint_recipe.py

```diff
@@ -149,9 +155,55 @@ def lintify(meta, recipe_dir=None):
     except RuntimeError as e:
         lints.append(str(e))
 
+    # 13: Check that the recipe_dir has the same name as the recipe_name
+    recipe_name = package_section.get('name', '').strip()
+    if recipe_dir:
```

Comment:
That explains it. I haven't seen this error in a while. I'll remove it. (Btw, for feedstocks, the `recipe_dirname != 'recipe'` is false and this check is not performed. See line 163)